src/main/resources/additional_properties.json

```diff
@@ -18,5 +18,266 @@
+                },
+                "whitelist" : {
+                  "type" : "array",
+                  "items" : {
+                    "type" : "string"
+                  }
+                },
+                "blacklist" : {
+                  "type" : "array",
+                  "items" : {
+                    "type" : "string"
+                  }
+                },
+                "renames" : {
+                  "type" : "array",
+                  "items" : {
+                    "type" : "string"
+                  }
+                }
+              }
+            }
+          }
+        },
+        {
+          "type" : "object",
+          "required" : [ "set_schema_metadata" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "set_schema_metadata" : {
+              "additionalProperties" : false,
+              "required" : [ "apply_to", "name", "version" ],
+              "properties" : {
+                "apply_to" : {
+                  "$ref" : "#/$defs/apply_to"
+                },
+                "name" : {
+                  "type" : "string",
+                  "default" : null
+                },
+                "version" : {
+                  "type" : "integer",
+                  "default" : null
+                }
+              }
+            }
+          }
+        },
+        {
+          "type" : "object",
+          "required" : [ "tombstone_handling" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "tombstone_handling" : {
+              "additionalProperties" : false,
+              "required" : [ "behaviour" ],
+              "properties" : {
+                "behaviour" : {
+                  "type" : "string",
+                  "default" : "warn",
+                  "enum" : [ "warn", "ignore", "fail" ]
+                }
+              }
+            }
+          }
+        },
+        {
+          "type" : "object",
+          "required" : [ "value_to_key" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "value_to_key" : {
+              "additionalProperties" : false,
+              "required" : [ "fields" ],
+              "properties" : {
+                "fields" : {
+                  "type" : "array",
+                  "items" : {
+                    "type" : "string"
+                  }
+                }
+              }
+            }
+          }
+        },
+        {
+          "type": "object",
+          "required": [ "topic_routing" ],
+          "additionalProperties": false,
+          "properties": {
+            "topic_routing": {
+              "additionalProperties": false,
+              "required": [ "topic_regex", "topic_replacement", "key_enforce_uniqueness", "key_field_name", "key_field_regex", "key_field_replacement" ],
+              "properties": {
+                "topic_regex": {
+                  "type": "string"
```

Comment:
I think there's a subtype for regex?
```
{
    "type": "string",
    "format": "regex"
}
```
Maybe `topic_names` makes more sense as a field name then?

Reply:
Yes, there is an option `pattern` to provide the regex pattern as well.

Reply:
I think `pattern` is for another use-case, isn't it? Maybe you could link to the paragraph in the JSON Schema docs that you were referring to?
I was referring to the built-in formats section: https://json-schema.org/understanding-json-schema/reference/string.html#built-in-formats which means there's `"format": "regex"` type for string types.